### ISSUE TITLE
Purpose of Batching in `pi_Ql`

### ISSUE THREAD

**Issue by ben-e-whitney** [CODARcode/MGARD#80]
`pi_Ql` interpolates function values at 'old' nodes, subtracting the results from the values at the 'new' nodes in between. For the 'innermost' 'new' nodes, [it does this in batches][batches]. @tugluk, is there a reason for this? It causes a bug when the grid is very small, as can be seen by compiling and executing the script below with the following commands (the central entry should be set to `0` but it remains `1`).

```console
git checkout 5630dc6c21d24dfbe22f323d07d6cf702fedc79d
make lib/libmgard.a
g++ -g -fno-omit-frame-pointer -O3 -march=native -ffast-math -fno-finite-math-only -std=c++11 -fPIC -Iinclude -c -o obj/missed.o missed.cpp
mkdir --parents bin
g++ -g -fno-omit-frame-pointer -O3 -march=native -ffast-math -fno-finite-math-only obj/missed.o lib/libmgard.a -lz -ldl -o bin/missed
```

```cpp
#include <iostream>

#include "mgard.h"

int main() {
  const std::size_t nrow = 3;
  const std::size_t ncol = 3;
  std::vector<float> v(nrow * ncol, 1);
  std::vector<float> row_vec(nrow);
  std::vector<float> col_vec(ncol);
  mgard::pi_Ql(nrow, ncol, 0, v.data(), row_vec, col_vec);
  std::vector<float>::const_iterator p = v.cbegin();
  for (std::size_t i = 0; i < nrow; ++i) {
    for (std::size_t j = 0; j < ncol; ++j) {
      std::cout << *p++ << " ";
    }
    std::cout << std::endl;
  }
  return 0;
}
```

[batches]: https://github.com/CODARcode/MGARD/blob/5630dc6c21d24dfbe22f323d07d6cf702fedc79d/src/mgard.tpp#L1374-L1401

**Comment by tugluk** [comment]
I remember doing this because this was the way I differentiated
between 'old' and 'new' nodes. This wouldn't work on 3x3 grid,  you're
right.

**Comment by ben-e-whitney** [comment]
This is fixed in [77f87be][commit] (my commit dated 2020-04-23T11:55:50-04:00).

[commit]: https://github.com/CODARcode/MGARD/commit/77f87be146d5cd491da9f01febc32364eaa1ae7a
